monitor.py

```diff
@@ -227,7 +228,12 @@ def all(cls, _db, **constructor_kwargs):
         service_match = or_(Timestamp.service==cls.SERVICE_NAME,
                             Timestamp.service==None)
         collections = Collection.by_protocol(_db, cls.PROTOCOL).outerjoin(
-            Collection.timestamps).filter(service_match)
+            Timestamp,
+            and_(
+                Timestamp.collection_id==Collection.id,
+                service_match,
+            )
+        )
```

Comment:
This was filtering out collections that had a Timestamp with a different service name. My change moves the filter into the on clause instead of the where clause, so it won't join Timestamps for other services, but the collections with those Timestamps will still be there.